libcnb-data/src/buildpack/id.rs

```diff
@@ -40,23 +40,46 @@ libcnb_newtype!(
     /// ```
     BuildpackId,
     BuildpackIdError,
-    r"^(?!app$|config$)[[:alnum:]./-]+$"
+    r"^(?!(app|config)$)[[:alnum:]./-]+$"
```

Comment:
This change is a no-op, and was to make the regex consistent with that for `LayerName`, and reduce chance of bugs, if a new entry is added here but the closing `$` omitted.